### ISSUE TITLE
scanpy.pp.filter_genes should have a batch key option

### ISSUE THREAD

**Issue by adamgayoso** [scverse/scanpy#992]
It would be great if `scanpy.pp.filter_genes` had a batch key option so that options like `min_counts` were taken with respect to each batch.

**Comment by ivirshup** [comment]
Would this be "genes where minimum counts in every batch is greater than `min_counts`"?

**Comment by adamgayoso** [comment]
Yes! I can't recall exactly but I think now what can happen is that after `filter_genes` and hvg selection using the batch key, you might end up with genes that have 0 counts in a particular batch, especially if picking the top G genes in hvg selection.

**Comment by snigdhaAgarwal** [comment]
I've started working on this. I was wondering if it's ok to change the signature of filter_genes so that it only accepts and returns anndata objects like hvg method?
